src/settings/SettingsComponent.cpp

```diff
@@ -305,6 +305,10 @@ void SettingsComponent::loadConf(const QString& path, bool storage)
       getSection(SETTINGS_SECTION_MAIN)->setValue("autodetectCertBundle", "true");
     }
   }
+
+#ifdef DISABLE_UPDATE_CHECK
+    getSection(SETTINGS_SECTION_MAIN)->setValueHidden("checkForUpdates", "true");
```

Comment:
Looks like this is forcing update checking to be enabled?

Reply:
This hides the value from the UI; the `true` is with regard to the `setValueHidden` method.
 **Before**
![Screenshot_20250402_142619](https://github.com/user-attachments/assets/4e811204-673f-4fff-8bd0-55a775eda68f)

**After**
![Screenshot_20250402_142935](https://github.com/user-attachments/assets/14a12847-adc0-4446-b1e9-3ab315f15c5b)